test/e2e/network/endpointslice.go

```diff
@@ -0,0 +1,428 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = SIGDescribe("EndpointSlice [feature:EndpointSlice]", func() {
+	version := "v1"
+	ginkgo.Context("version "+version, func() {
+		f := framework.NewDefaultFramework("endpointslice")
+
+		var cs clientset.Interface
+		var podClient *framework.PodClient
+
+		ginkgo.BeforeEach(func() {
+			cs = f.ClientSet
+			podClient = f.PodClient()
+		})
+
+		labelPod1 := "pod1"
+		labelPod2 := "pod2"
+		labelPod3 := "pod3"
+		labelShared12 := "shared12"
+		labelValue := "on"
+
+		ginkgo.It("should create Endpoints and EndpointSlices for Pods matching a Service", func() {
+			svc1 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-int-port",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelPod1: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "example",
+						Port:       80,
+						TargetPort: intstr.FromInt(3000),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			svc2 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-named-port",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelShared12: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "http",
+						Port:       80,
+						TargetPort: intstr.FromString("example-name"),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			svc3 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "example-no-match",
+				},
+				Spec: v1.ServiceSpec{
+					Selector: map[string]string{labelPod3: labelValue},
+					Ports: []v1.ServicePort{{
+						Name:       "example-no-match",
+						Port:       80,
+						TargetPort: intstr.FromInt(8080),
+						Protocol:   v1.ProtocolTCP,
+					}},
+				},
+			})
+
+			pod1 := podClient.Create(&v1.Pod{
```

Comment:
nit: create pods before services.
That might be make the test slightly faster

Reply:
Good catch, that does seem to help.